src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/OrderLimitPriceField.tsx

```diff
@@ -110,33 +113,40 @@ export function OrderLimitPriceField({ id }: OrderLimitPriceFieldProps) {
       // When price is below or equal to 0, set the limit price to 0, but don't update the state
 
       // get and parse the sell token amount
-      const sellTokenAmountFloat = parseFloat(
-        formatUnits(sellTokenAmount.raw.toString(), sellTokenAmount.currency.decimals)
-      )
 
-      let newBuyAmountAsFloat = 0 // the amount of buy token
-
-      if (limitOrder.kind === LimitOrderKind.SELL) {
-        newBuyAmountAsFloat = sellTokenAmountFloat * nextLimitPriceFloat
+      if (inputFocus === InputFocus.SELL) {
+        const {
+          amount: newBuyAmountFloat,
+          buyAmountWei: nextBuyAmountWei,
+          newBuyTokenAmount,
+        } = computeNewAmount(buyTokenAmount, sellTokenAmount, nextLimitPriceFloat, limitOrder.kind, inputFocus)
+
+        setBuyTokenAmount(newBuyTokenAmount)
+        setFormattedBuyAmount(newBuyAmountFloat.toString())
+
+        setFormattedLimitPrice(nextLimitPriceFormatted)
+        setLimitOrder({
+          ...limitOrder,
+          limitPrice: parseUnits(nextLimitPriceFormatted, quoteTokenAmount?.currency?.decimals).toString(),
+          buyAmount: nextBuyAmountWei,
+        })
       } else {
-        newBuyAmountAsFloat = sellTokenAmountFloat / nextLimitPriceFloat
+        const {
+          amount: newSellAmountFloat,
+          sellAmountWei: nextSellAmountWei,
+          newSellTokenAmount,
+        } = computeNewAmount(buyTokenAmount, sellTokenAmount, nextLimitPriceFloat, limitOrder.kind, inputFocus)
```

Comment:
This function is invoked twice with the same inputs. Take it out and destruct everything required.

Reply:
Thanks, forgot about it. I built this function exactly for this!